crates/bevy_math/src/primitives/polygon.rs

```diff
@@ -0,0 +1,340 @@
+extern crate alloc;
+use alloc::collections::BTreeMap;
+use core::cmp::Ordering;
+
+use crate::Vec2;
+
+use super::{Measured2d, Triangle2d};
+
+#[derive(Debug, Clone, Copy)]
+enum Endpoint {
+    Left,
+    Right,
+}
+
+/// An event in the [`EventQueue`] is either the left or right vertex of an edge of the polygon.
+///
+/// Events are ordered so that any event `e1` which is to the left of another event `e2` is less than that event.
+/// If `e1.position().x == e2.position().x` the events are ordered from bottom to top.
+#[derive(Debug, Clone, Copy)]
+struct Event {
```

Comment:
I know this is not exposed outside the module, but I feel that even inside the module this would deserve a smaller scope, or a more concrete name than just "Event".

Reply:
Something like `SweepLineEvent`?